bullet_train-super_scaffolding/lib/generators/super_scaffold/super_scaffold_generator.rb

```diff
@@ -0,0 +1,49 @@
+class SuperScaffoldGenerator < Rails::Generators::NamedBase
+  source_root File.expand_path("templates", __dir__)
+
+  # ##############################
+  #
+  # TODO: Should we retain some of these class options?
+  #
+  # By default we inherit a bunch of options that _would_ automatically be used if we
+  # were writing a normal Rails generator that used normal methods of generating files.
+  # So if you invoke this generator without passing any arguments (and without these
+  # remove_class_option lines) then at the top of the help text you'd see info about all
+  # of these options:
+  #
+  # ##############################
+  #
+  # rails g super_scaffold
+  #
+  # Options:
+  #   [--skip-namespace], [--no-skip-namespace]              # Skip namespace (affects only isolated engines)
+  #   [--skip-collision-check], [--no-skip-collision-check]  # Skip collision check
+  #
+  # Runtime options:
+  #   -f, [--force]                    # Overwrite files that already exist
+  #   -p, [--pretend], [--no-pretend]  # Run but do not make any changes
+  #   -q, [--quiet], [--no-quiet]      # Suppress status output
+  #   -s, [--skip], [--no-skip]        # Skip files that already exist
+  #
+  # ##############################
+  #
+  # Maye we have comparable options for some of these and should retain them and
+  # pass them through to the rake task?
+
+  remove_class_option :skip_namespace
+  remove_class_option :skip_collision_check
+  remove_class_option :force
+  remove_class_option :pretend
+  remove_class_option :quiet
+  remove_class_option :skip
+
+  def generate
+    # In order to pass the args to the rake task we have to join them
+    # into a single string with spaces and commas escaped. (Due to
+    # weird rake syntax requirements.) We add the name of the specific
+    # super_scaffolding command that we want to invoke to the beginning
+    # of the argument string.
+    new_args = (['crud'] + ARGV).join("\\ ").gsub(",", "\\,")
+    rake "bullet_train:super_scaffolding[#{new_args}]"
```

Comment:
Another option, although it's more work, is to flip the order so the rake task is an alternative invocation for the generators.

Kinda something like this?

```sh
#!/usr/bin/env ruby

require "shellwords"

command = "super_scaffold"

if %w[crud field join-model oauth-provider incoming-webhooks].include?(ARGV.first)
 command += ":#{ARGV.shift}"
end

system "bin/rails generate #{command} #{Shellwords.escape(ARGV)}"
```

Then we could retain the class options like you're talking about.

Another benefit to using generators is that we can leverage `create_file` which will automatically generate a diff in case there's already a file at that path.

Reply:
I think this is an option that we should eventually take, but I'm not sure we want to go straight for it. I think it would involve quite a large refactoring of the code generation process. Using Rails generator helpers like `create_file` would definitely buy us a lot of extra functionality, like diffs and automatic support for `--force` and `--pretend` and what not.

Reply:
Or maybe we go ahead and start with the simple version that you've suggested here and just put off the use of `create_file` and friends for a later PR... 🤔

Reply:
@jagthedrummer not sure how I've missed your messages in this thread, but everything you said sounds good to me! And 👍 to taking it slow if we want to convert how super scaffolding executes.